File: Aromata/Aromata.Domain/Recipes/Recipe.cs
using Aromata.Domain.Common;

namespace Aromata.Domain.Recipes;

public class Recipe : BaseAuditableEntity
{
    public Guid BookId { get; set; }
    public string? Title { get; set; }
    public int? Page { get; set; }
    public string? Category { get; set; }
    
    
}

File: aromata_frontend/lib/models/recipe.dart
class Recipe {
  final String? id;
  final String bookId;
  final String title;
  final int page;
  final List<String> tags;

  Recipe({
    this.id,
    required this.bookId,
    required this.title,
    required this.page,
    this.tags = const [],
  });

  Recipe copyWith({
    String? id,
    String? bookId,
    String? title,
    int? page,
    List<String>? tags,
  }) {
    return Recipe(
      id: id ?? this.id,
      bookId: bookId ?? this.bookId,
      title: title ?? this.title,
      page: page ?? this.page,
      tags: tags ?? this.tags,
    );
  }
}

